### PR TITLE
Adding way to include/exclude specific models by specifying app_label

### PR DESCRIPTION
**Pull request by AbhiAgarwal**
The two last pull requests I sent were conflicting. Sorry I had to send a new one:

The project I am working on has a large code-base and thus has models which have the same name. The current method only filters the models by name, and not by the app they are in. This is more for django applications with multiple models.

I needed to add the ability to specify which app the particular model belongs to. I have tested it and I don't think this particular method breaks anything. The simple change would be the ability to add demo_models.foo rather than foo. I kept it backwards compatible and so if there's foo it would still work.

Let me know if you've any feedback. I would be happy to add/edit this particular solution.

### CODE REVIEW

File: report_builder/models.py [hunk 1]
```diff
@@ -21,9 +21,26 @@
...
     if getattr(settings, 'REPORT_BUILDER_INCLUDE', False):
-        models = models.filter(model__in=settings.REPORT_BUILDER_INCLUDE)
+        allModelNames = ()
+        additionalModels = []
+        for element in settings.REPORT_BUILDER_INCLUDE:
+            splitElement = element.split('.')
+            if len(splitElement) == 2:
+                additionalModels.append(models.filter(app_label=splitElement[0], model=splitElement[1]))
+            else:
+                allModelNames = allModelNames + (element,)
```
Comment (grantjenks) [inline report_builder/models.py]: This is a very slow way to add items to a list. It is much more efficient to use `append` with a list than summing tuples together.

**Comment by AbhiAgarwal** [comment]
@grantjenks That makes sense. I completely blanked on tuples being immutable. I've made that simple change. Let me know if anything else pops up. I was a little worried about efficiency here and I tried to design the change in `REPORT_BUILDER_INCLUDE` in the best way I could think of.

**Comment by bufke** [comment]
Hello this looks good but needs a unit test. Once you add a unit test that would fail now and succeed with this pull I will merge. I have to do this because I will accidentally break this later without a test :) Testing against just the get_allowed_models function would be sufficient.

**Comment by AbhiAgarwal** [comment]
@bufke I manage to add test called `test_get_allowed_models`. I had to add a separate app to test this (to test for duplicate models that would create between different apps).

I had to make some changes with a couple existing tests as well. I had to specify the `app_label` as now we have two separate "bars" (in `demo_models`, and `demo_second_app`.

**Comment by grantjenks** [comment]
How does your test work? I don't see REPORT_BUILDER_INCLUDE and REPORT_BUILDER_EXCLUDE set or modified in test_get_allowed_models.

**Comment by AbhiAgarwal** [comment]
@grantjenks In the function [here](https://github.com/AbhiAgarwal/django-report-builder/commit/9c3545b950ff5673163f082383717e375790c497#diff-f8e0c1d6b64ce8c6942e89b2394836eaR113) I set the `REPORT_BUILDER_INCLUDE`. 

The test being:

```python
def test_get_allowed_models(self):
    preIncludeDuplicates = find_duplicates_in_contexttype()
    settings.REPORT_BUILDER_INCLUDE = (
        'demo_models.bar',
        'demo_models.foo',
        'demo_models.place',
    )
    postIncludeDuplicates = find_duplicates_in_contexttype()
    self.assertEqual(preIncludeDuplicates, ['bar'])
    self.assertEqual(postIncludeDuplicates, [])
```

**Comment by bufke** [comment]
I'm going to be picky and ask that you use a consistent coding style. No camelCase please. Use stuff_like_this. 

Also I think we app_label.model is better than just model. Let's change the docs to say only app_label.model and then just silently support having only the model for compatibility reasons.

**Comment by AbhiAgarwal** [comment]
@bufke Thank you for the comments! I totally dropped the ball on the Python style there. I have fixed that for this release! I had to make some changes to the `tests.py` as we started to include models with the same names, and so I added app_label for each model in the tests as well. 

I added the docs to reflect that change as well - I made the examples specify a model along with an app_label.

**Comment by saulshanabrook** [comment]
@AbhiAgarwal Could you also [squash all your commits](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) to clean up the history?

**Comment by AbhiAgarwal** [comment]
@saulshanabrook yup! I just did it here :)
